rcl_logging_log4cxx/src/rcl_logging_log4cxx/rcl_logging_log4cxx.cpp

```diff
@@ -102,13 +102,12 @@ rcl_logging_ret_t rcl_logging_external_initialize(
     // To be compatible with ROS 1, we want to construct a default filename of
     // the form ~/.ros/log/<exe>_<pid>_<milliseconds-since-epoch>.log
 
-    // First get the home directory.
-    const char * homedir = rcutils_get_home_dir();
-    if (homedir == nullptr) {
-      // We couldn't get the home directory; it is not really going to be
-      // possible to do logging properly, so get out of here without setting
-      // up logging.
-      return RCL_LOGGING_RET_ERROR;
+    char * logdir = nullptr;
+    rcl_logging_ret_t dir_ret = rcl_logging_get_logging_directory(allocator, &logdir);
```

Comment:
Down below this line, if there are failures (like `rcutils_system_time_now`), we are leaking memory.

Reply:
fixed as part of dd0a9768e2fd4ad15c087c849c090e4e1364509b